src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -16,222 +16,75 @@
-	 *
-	 */
-	void set_dim_feature_space(const int32_t dim);
-
-	/** computes new random coefficients IF test_rfinited() evaluates to false
-	 * test_rfinited() evaluates to TRUE if void set_randomcoefficients(...)
-	 * hase been called and the values set by set_dim_input_space(...) ,
-	 * set_dim_feature_space(...) and set_kernelwidth(...) are consistent to the
-	 * call of void set_randomcoefficients(...)
+	/** @brief Preprocessor that approximates Gaussian feature map.
 	 *
-	 * throws shogun exception if dim_feature_space <= 0 or dim_input_space <= 0
+	 * Gaussian kernel of form K(x,y) = exp(-||x - y||^2 / kernelwidth)
+	 * is approximated using the formula: 
 	 *
-	 * @return returns true if test_rfinited() evaluates to false and new
-	 * coefficients are computed
-	 * returns false if test_rfinited() evaluates to true and old random
-	 * coefficients are kept which were set by a previous call to void
-	 * set_randomcoefficients(...)
-	 *
-	 * this function is useful if you want to use apply_to_feature_vector but
-	 * cannot call before it fit(Features *f)
+	 * \f[
+	 * 	z(x) = \sqrt{2/D}\ cos(w'x + b)
+	 *  K(x, y) = \langle\phi(x),\phi(y)\rangle \approx z'(x)z(y)
+	 * \f]
+	 * 
+	 * Reference:
+	 * [1] Rahimi, A., & Recht, B. (2008). Random features for large-scale kernel
+	 * machines. In Advances in neural information processing systems (pp.
+	 * 1177-1184).
 	 *
 	 */
-	bool init_randomcoefficients();
+class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>> {
+public:
+		/** default constructor */
+		RandomFourierGaussPreproc();
 
-	/** a getter
-	 * @return the set value of protected member dim_input_space
-	 */
-	int32_t get_dim_input_space() const;
+		~RandomFourierGaussPreproc();
 
-	/** a getter
-	 * @return the set value of protected member dim_feature_space
-	 */
-	int32_t get_dim_feature_space() const;
+		virtual SGVector<float64_t> apply_to_feature_vector(SGVector<float64_t> vector) override;
 
-	/** inherited from base class
-	 * does nothing
-	 */
-	void cleanup();
+		/** @return C_DENSE*/
+		virtual EFeatureType get_feature_type() override;
 
-	/// return the name of the preprocessor
-	virtual const char* get_name() const { return "RandomFourierGaussPreproc"; }
+		/** @return F_DREAL */
+		virtual EFeatureClass get_feature_class() override;
 
-	/// return a type of preprocessor
-	virtual EPreprocessorType get_type() const { return P_RANDOMFOURIERGAUSS; }
+		virtual void fit(std::shared_ptr<Features> f) override;
 
-protected:
-	/** default processing routine, inherited from base class
-	 * @param matrix the features matrix to be processed
-	 * @return the processed feature matrix from the DenseFeatures<float64_t>
-	 * class in case (2) (see description above) this routine requires only
-	 * steps 2a) and 2b), the rest is determined automatically
-	 */
-	virtual SGMatrix<float64_t> apply_to_matrix(SGMatrix<float64_t> matrix);
+		/** does nothing */
+		void cleanup();
```

Comment:
if the method does nothing, remove it (and yes also in the base class)

Reply:
It does nothing here, but in the base class(and other derivatives) it resets the internal members. 
If we plan to remove the method despite that, we'd need to remove from all the derivatives which I think is a bit out of the scope of PR?

Reply:
Ah yes, I forgot. Just leave it as it is then (for now)